### DISCUSSION TITLE
对于实现连续对话的思考

### DISCUSSION THREAD

**Discussion by wiketool** [nonebot/discussions#404] [问答]
现在我需要开发一个功能，功能核心流程大概是以下
1. 用户发送`/START`后开始处理流程
2. bot可以连续接收用户发送的文字或图片
3. 直到用户发送`/EOF`后将这些文字或图片组装起来
4. 结束该流程

这个功能主要涉及bot的连续对话，也就是第2步，在借鉴了几个用nonebot实现的chatgpt后，我对这个功能有下面几种不同的实现思路：
1. 设置三个matcher，分别对应1、2、3步，第一步会下发一个uuid作为本次流程的编号，将会话的状态保存到本地或内存
2. 设置一个matcher，matcher里判断有无关键字从而确定是第几步的操作【如果当前用户不处于流程中，则pass】，和第一种想法大同小异
3. 设置一个matcher对应第一步，之后依靠添加临时matcher的方式实现连续对话和流程控制

目前我只想到了这三种思路实现，就当我抛砖引玉，想问下有没有其他的实现思路？

**Comment by yanyongyu** [comment]
只需要定义一个matcher，然后在接收图片的时候不断reject直到收到eof即可

**Reply by wiketool** [reply]
Nice idea!这个想法比我那几个好多了
